sourcing/domain/service.go

```diff
@@ -452,9 +454,11 @@ func (d *DefaultSourcingService) getSourceStock(
 func (s AvailableSources) QtySum() int {
 	qty := 0
 	for _, sqty := range s {
-		if sqty == math.MaxInt64 {
-			return sqty
+		// check against max int 32 to avoid overflowing int
+		if sqty == MaxSourceQyt || qty > math.MaxInt32 {
```

Comment:
can you explain the second check for max int 32?

Reply:
package main

import (
	"fmt"
	"math"
)

func main() {
	sourceqty := int64(math.MaxInt64) - int64(1)
	sourceqty1 := int64(math.MaxInt64) - int64(1)

	sum := sourceqty + sourceqty1

	fmt.Println(sum) // -4
}

if fore some reason sqource qty will be almost maxint64 and not exactly and we sum it up we will have a wierd number like -4 in this example